File: tests/fixtures/kotlin/example-auth0/src/main/kotlin/com/example/AuthController.kt
package com.example

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.servlet.view.RedirectView
import jakarta.servlet.http.HttpSession
import jakarta.servlet.http.HttpServletRequest

@Controller
class AuthController(
    @Value("\${auth0.domain}") private val auth0Domain: String,
    @Value("\${auth0.clientId}") private val clientId: String,
    @Value("\${auth0.clientSecret}") private val clientSecret: String,
) {
    @GetMapping("/api/health")
    @org.springframework.web.bind.annotation.ResponseBody
    fun health(): Map<String, String> {
        return mapOf("status" to "ok", "version" to "1.0.0")
    }

    @GetMapping("/login")
    fun login(): RedirectView {
        val authorizeUrl = "https://$auth0Domain/authorize?" +
            "client_id=$clientId&redirect_uri=http://localhost:8080/callback&response_type=code&scope=openid profile email"
        return RedirectView(authorizeUrl)
    }

    @GetMapping("/callback")
    fun callback(request: HttpServletRequest, session: HttpSession): RedirectView {
        val code = request.getParameter("code")
        // In a real app, exchange code for tokens via Auth0 API
        session.setAttribute("user", mapOf("code" to code))
        return RedirectView("/")
    }

    @GetMapping("/logout")
    fun logout(session: HttpSession): RedirectView {
        session.invalidate()
        return RedirectView("/")
    }
}

File: tests/fixtures/php-laravel/example-auth0/app/Http/Controllers/AuthController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Auth0\Laravel\Facade\Auth0;

class AuthController extends Controller
{
    public function home()
    {
        $user = Auth0::getUser();
        return view('home', ['user' => $user]);
    }

    public function login()
    {
        return Auth0::login();
    }

    public function callback()
    {
        Auth0::callback();
        return redirect('/');
    }

    public function logout()
    {
        Auth0::logout();
        return redirect('/');
    }
}
